src/main/java/eu/ai4work/sws/service/RuleEngineService.java

```diff
@@ -103,4 +113,53 @@ private void setInputParametersToFuzzyInterfaceSystem(FIS fuzzyInferenceSystem,
             }
         });
     }
+
+    /**
+     * Build a basic explanation of the sliding decision by showing membership values and rules that were applied.
+     *
+     * @param fuzzyInferenceSystem the fuzzy inference system to obtain the explanation.
+     * @return a structured Map containing the overall explanation.
+     */
+    private Map<String, Object> buildSlidingDecisionExplanation(FIS fuzzyInferenceSystem) {
+        var functionBlock = fuzzyInferenceSystem.getFunctionBlock(null);
+
+        return Map.of(
```

Comment:
Although such a map is very generic and works fine, it is a little bit hard to understand its structure. I am wondering if it may be better  to use a structured object at least on those levels that are application-scenario-agnostic, e.g.:

```
@RequiredArgsConstructor
@Data
public class SlidingDecisionExplanation {

    private final List<FuzzyVariableExplanation> inputVariables;
    private final List<RuleExplanation> appliedRules;
    private final List<FuzzyVariableExplanation> outputVariables;
}
```